apps/projects/tasks.py

```diff
@@ -0,0 +1,144 @@
+
+        # Use tool for editing video
+        video_editor = get_video_editor()
+        edited_video_stream, metadata = video_editor.edit_video(
+            video_stream,
+            doc['filename'],
+            doc.get('metadata'),
+            updates.get('cut'),
+            updates.get('crop'),
+            updates.get('rotate'),
+            updates.get('quality')
+        )
+        if action == 'post':
+            new_storage_id = app.fs.put(edited_video_stream, doc.get('filename'), None)
+        elif action == 'put':
+            new_storage_id = app.fs.replace(edited_video_stream, storage_id, None)
+        else:
+            raise KeyError(f'Invalid action `{action}`')
+
+        # create url for preview video
+        url = app.fs.url_for_media(doc.get('_id'))
+        # Update data status is True and data video when edit was finished
+        app.mongo.db.projects.find_one_and_update(
+            {'_id': doc['_id']},
+            {'$set': {
+                'processing': False,
+                'metadata': metadata,
+                'storage_id': new_storage_id,
+                'thumbnails': {},
+                'url': url
+            }},
+            return_document=ReturnDocument.AFTER
+        )
+        # Delete all old thumbnails
+        thumbnails = []
+        for key in doc['thumbnails'].keys():
+            thumbnails = doc['thumbnails'][str(key)]
+        for thumbnail in thumbnails:
+            app.fs.delete(thumbnail['storage_id'])
+
+    except Exception as exc:
+        logger.exception(exc)
+        if retry < app.config.get('NUMBER_RETRY', 3):
+            task_edit_video.delay(json_util.dumps(doc), updates, action, retry=retry + 1)
+        else:
+            if doc['version'] >= 2:
+                app.mongo.db.projects.delete_one({'_id': doc['_id']})
+            else:
+                app.mongo.db.projects.update_one(
+                    {'_id': doc['_id']},
+                    {'$set': {
+                        'processing': False,
+                    }}
+                )
+
+
+@celery.task
+def task_get_list_thumbnails(sdoc, amount, retry=0):
+    update_thumbnails = []
+    try:
+        doc = json_util.loads(sdoc)
+
+        # get full path file of video
+        filename, ext = os.path.splitext(doc['filename'])
+        stream_file = app.fs.get(doc['storage_id'])
+        video_editor = get_video_editor()
+        count = 0
+        for thumbnail_stream, \
+            thumbnail_meta in video_editor.capture_list_timeline_thumbnails(stream_file,
+                                                                            filename,
+                                                                            doc.get('metadata'),
+                                                                            int(amount)):
+            thumbnail_filename = '%s_timeline_%02d.png' % (filename, count)
+            storage_id = app.fs.put(thumbnail_stream, thumbnail_filename, 'image/png')
```

Comment:
I think we need more parameters in `fs.put` to archive that. Because we generate `storage_id` inside `fs.put`, so there is no way for knowing where video and thumbnails are stored, my suggestion:
1. `project_id`
2. `type`: to identify which type of assets are, if `type` is video, we will store it in `year/month/day/project_id/filename`, and `year/month/day/project_id/type/filename` otherwise.
3. `storage_id` (of video): to know where video is stored without query database